android/src/main/java/com/ryanheise/just_audio/MethodCallHandlerImpl.java

```diff
@@ -0,0 +1,52 @@
+package com.ryanheise.just_audio;
+
+import android.content.Context;
+import android.util.SparseArray;
+import androidx.annotation.NonNull;
+import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
+import io.flutter.plugin.common.MethodChannel.Result;
+
+public class MethodCallHandlerImpl implements MethodCallHandler {
+
+	private final Context applicationContext;
+	private final BinaryMessenger messenger;
+
+	private final SparseArray<AudioPlayer> players = new SparseArray<>();
+
+	public MethodCallHandlerImpl(Context applicationContext,
+			BinaryMessenger messenger) {
+		this.applicationContext = applicationContext;
+		this.messenger = messenger;
+	}
+
+	@Override
+	public void onMethodCall(MethodCall call, @NonNull Result result) {
+		switch (call.method) {
+		case "init":
+			int id = Integer.parseInt(call.argument("id"));
+			players.put(id, new AudioPlayer(applicationContext, messenger, id,
+					() -> players.remove(id)
+			));
+			result.success(null);
+			break;
+		case "setIosCategory":
+			result.success(null);
+			break;
+		default:
+			result.notImplemented();
+			break;
+		}
+	}
+
+	void dispose() {
+		final int size = players.size();
+		for (int i = 0; i < size; i++) {
+			final AudioPlayer player = players.valueAt(i);
+			player.dispose();
```

Comment:
This will cause the size of the array to become shorter while iterating, and may cause every second player to be skipped and not get disposed. You might want to iterate over a clone of the array to iterate over a stable collection.  If `dispose` triggers the runnable `players.remove(id)` to be called, then also the `players.clear()` line below is redundant.

Reply:
Ok I forgot about that. Will update..

Reply:
The `players.clear()` line below is still redundant, though.

Reply:
Also, `players.values()` does not make a stable clone of the map. It's actually backed by the map, so any changes to the map will appear in `values`.